test/integration/suites/upgrade/02-verify-codebase-version-is-updated

```diff
@@ -1,40 +1,60 @@
 #!/bin/bash
 
-check-version-against-prior-tag() {
-    # Get nearest prior tag in this branch
-    _nearest_prior_tag=$(git --git-dir $REPODIR/.git describe --match "v*" --abbrev=0 | cut -c 2-)
+check-version-against-latest-release() {
+    # This should be set by the CI/CD system based on the target branch of
+    # the PR.
+    if [ -n "${TEST_TARGET_BRANCH}" ]; then
+        log-info "target branch (explicit): ${TEST_TARGET_BRANCH}"
+    else
+        TEST_TARGET_BRANCH="$(git --git-dir "${REPODIR}"/.git branch --show-current)"
+        log-info "target branch (detected): ${TEST_TARGET_BRANCH}"
+    fi
 
-    if [ "$_nearest_prior_tag" == "$1" ]; then
-        fail-now "codebase version is the same than the last tag ($1), was the codebase version number bumped after the last release?"
+    _commit_version="$1"
+
+    if [ "${TEST_TARGET_BRANCH}" = "master" ]; then
```

Comment:
re: #2003 , should we try to discover the default branch name to avoid some rework later? Something like:
```suggestion
    _default_branch=$(git remote show origin | grep 'HEAD branch' | cut -d ":" -f 2 | xargs)
    if [ "${TEST_TARGET_BRANCH}" = "${_default_branch}" ]; then
```

Reply:
Heh, sounds good.